runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -351,7 +351,15 @@ public void onChatMessage(ChatMessage event)
 
 		if (config.screenshotKills() && KILL_MESSAGES.stream().anyMatch(chatMessage::contains))
 		{
-			String fileName = "Kill " + format(new Date());
+			String kill_name = chatMessage;
+
+			for(String test_string : KILL_MESSAGES) {
+				kill_name = kill_name.replace(test_string, "");
```

Comment:
Some of the kill messages are missing more than just the name. Ex. "Can anyone defeat you? Certainly not NAME"

Reply:
This one too: `"into tiny pieces and sat on them"`

Reply:
> This one too: `"into tiny pieces and sat on them"`

This one is already in the `KILL_MESSAGES`. 

Is there a list of PvP messages? Will go through and make sure all are covered if needed.

Reply:
The issue is that your string replace doesn't worry correctly with some messages for example: `"into tiny pieces and sat on them"` is only the second part of the string, the entire string is `You cut {username} into tiny pieces and sat on them`, for example. 

That would make the filename `Kill - You cut username - {DATE}`

Reply:
True, however, if I could get ahold of all the messages I could write them all out ommitting usernames. In theory this would call the method once but remove all parts of the message.